src/main/src/main/java/org/geoserver/catalog/StoreInfo.java

```diff
@@ -170,4 +170,18 @@ public interface StoreInfo extends CatalogInfo {
      */
     // <T extends Resource> Iterator<T> getResources(ProgressListener monitor) throws IOException;
 
+    /**
+     * Get the auto disable flag on connection failure.
```

Comment:
```suggestion
     * When true, the {@link ResourcePool} will automatically disable the store on connection failure. Defaults to false.
```

(I should have linked to the Resource